File: plot.m
% Path to the CSV file
csv_file_path = 'heart_rate_data.csv';

% Load the data
data = readtable(csv_file_path, 'ReadVariableNames', false);
heart_rate = data.Var1; % Assumes the heart rate data is in the first column

% Plot the data
figure;
plot(heart_rate, 'b', 'DisplayName', 'Heart Rate'); % 'b' stands for blue color
title('Heart Rate Data');
xlabel('Sample Number');
ylabel('Heart Rate Value');
legend;
grid on;

File: plot.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Feb  4 09:04:27 2024

@author: saifnoreldeen
"""

import pandas as pd
import matplotlib.pyplot as plt

# Path to the CSV file
csv_file_path = 'heart_rate_data.csv'

def plot_heart_rate_data(csv_file_path):
    # Load the data
    data = pd.read_csv(csv_file_path, header=None, names=['Heart Rate'])

    # Plot the data
    plt.figure(figsize=(10, 6))
    plt.plot(data['Heart Rate'], label='Heart Rate', color='blue')
    plt.title('Heart Rate Data')
    plt.xlabel('Sample Number')
    plt.ylabel('Heart Rate Value')
    plt.legend()
    plt.grid(True)
    plt.show()

if __name__ == "__main__":
    plot_heart_rate_data(csv_file_path)
